tests/e2e/utils/track-event.js

```diff
@@ -10,14 +10,27 @@
  *
  * @param {Page} page
  * @param {string} eventName Event name to match.
+ * @param {string|null} URLpath The URL path where the event should be triggered.
  * @return {Promise<Request>} Matching request.
  */
-export function trackGtagEvent( page, eventName ) {
+export function trackGtagEvent( page, eventName, URLpath = null ) {
 	const eventPath = '/pagead/';
 	return page.waitForRequest( ( request ) => {
 		const url = request.url();
 		const match = encodeURIComponent( 'event=' + eventName );
-		return url.includes( eventPath ) && url.includes( match );
+		const origin = new URL( page.url() ).origin;
+
+		return (
+			url.includes( eventPath ) &&
+			url.includes( match ) &&
+			( URLpath
+				? url.includes(
+						`url=${ encodeURIComponent(
+							`${ origin }/${ URLpath }`
+						) }`
+				  )
+				: true )
+		);
```

Comment:
I'm not an expert in JS code but it takes me a while to grok what's being compared here.
I wonder if it would be simpler to convert the request URL to a [URL object](https://developer.mozilla.org/en-US/docs/Web/API/URL) so we can check if the event `searchParams` is set and matches the eventName.

That would also allow for more accurate matching with the exact URL. Looking at the payload of a request I see:
![image](https://github.com/woocommerce/google-listings-and-ads/assets/11388669/cee910cf-e4c7-4d10-9bf9-03ba5b3c97d2)

So using the `url.includes` might not be very foolproof as it could easily match the referrer URL which would be set to the previous page visited.

Reply:
Thanks, @mikkamp, for the suggestion. I don't believe that scenario would occur because I added the prefix `url=` within `url.includes`. Nonetheless, I've made the switch to using URL parameters for the sake of clarity.

Reply:
Ah, thanks for clarifying. I seem to have missed the `url=` prefix you added.

Since we now extract the params does it also mean we can change `eventName === params.get( 'event' )`

Reply:
I don't think so because the `event` is not a parameter, it is included in the `data` parameter.

![image](https://github.com/woocommerce/google-listings-and-ads/assets/2488994/caf3c526-27af-4e9c-8c73-308b00478f35)

We could do something like: `params.get( 'data' ).includes(match)` if you believe it's a slightly more clear compared to the current implementation.

Reply:
Ok, that makes sense. I do think the last suggestion you made does make it quite explicit what we are expecting to match (which might help with troubleshooting).

Reply:
Yep that sounds right. I made the change here: https://github.com/woocommerce/google-listings-and-ads/pull/2122/commits/6557bd19f84999acae74ccc7c4074bf2b51f2a1b